spec/p2p/reactor/p2p-api.md

```diff
@@ -0,0 +1,276 @@
+# API for Reactors
+
+This document describes the API provided by the p2p layer to the protocol
+layer, namely to the registered reactors.
+
+This API consists of two interfaces: the one provided by the `Switch` instance,
+and the ones provides by multiple `Peer` instances, one per connected peer.
+The `Switch` instance is provided to every reactor as part of the reactor's
+[registration procedure][reactor-registration].
+The multiple `Peer` instances are provided to every registered reactor whenever
+a [new connection with a peer][reactor-addpeer] is established.
+
+
+## `Switch` API
+
+The [`Switch`][switch-type] is the central component of the p2p layer
+implementation.  It manages all the reactors running in a node and keeps track
+of the connections with peers.
+The table below summarizes the interaction of the standard reactors with the `Switch`:
+
+| `Switch` API method                                     | consensus | block sync | state sync | mempool | evidence  | PEX   |
+|--------------------------------------------|-----------|------------|------------|---------|-----------|-------|
+| `Peers() IPeerSet`                         | x         | x          |            |         |           | x     |
+| `NumPeers() (int, int, int)`               |           | x          |            |         |           | x     |
+| `Broadcast(Envelope) chan bool`            | x         | x          | x          |         |           |       |
+| `MarkPeerAsGood(Peer)`                     | x         |            |            |         |           |       |
+| `StopPeerForError(Peer, interface{})`      | x         | x          | x          | x       | x         | x     |
+| `StopPeerGracefully(Peer)`                 |           |            |            |         |           | x     |
+| `Reactor(string) Reactor`                  |           | x          |            |         |           |       |
```

Comment:
Similar to @josef-widder 's comment above, do we want to keep this method moving forward? (it seems to enable implicit assumptions on the reactors registered, thus hindering modularity)

Reply:
I see you're explaining further below that its usage is discouraged. I think an output of all these spec-related PRs could a list of things we don't like, which can then be turned into a tracking issue that we can prioritize and plan as tech-debt.

Reply:
It is basically the idea to collect all the points we don't like or need to discuss whether we like them in https://github.com/cometbft/knowledge-base/pull/8.